content/build/ci/github-actions/attestations.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Add SBOM and provenance attestations with GitHub Actions
+description: Add SBOM and provenance attestations to your images with GitHub Actions
+keywords: ci, github actions, gha, buildkit, buildx, attestations, sbom, provenance, slsa
+---
+
+SBOM and provenance [attestations](../../attestations/_index.md) add valuable
+metadata about the contents of your image, and how it was built. Adding these
+attestations to your images with GitHub Actions is as simple as adding two
+input fields: `provenance` and `sbom`.
+
+```yaml
+name: ci
+
+on:
+  push:
+    branches:
+      - "main"
+
+env:
+  IMAGE_NAME: user/app
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Extract metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.IMAGE_NAME }}
+
+      - name: Build and push image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          sbom: true
+          provenance: true
```

Comment:
Worth checking if this defaults to `mode=max` or `mode=min`. For most of what we need for docker scout, we need `max`. Previously this defaulted to min because of security concerns capturing environment variables that may include secrets, but we may have changed this default. There may be additional steps to set it to mode=max.

Reply:
We set max provenance for public repositories but min for private repositories to avoid leaking sensitive information if the image is pushed on a public registry in our build-push-action: https://github.com/docker/build-push-action/blob/4a13e500e55cf31b7a5d59a38ab2040ab0f42f56/src/context.ts#L153-L168

We also automatically sets the builder id according to the workflow environment: https://github.com/docker/actions-toolkit/blob/64a550a5dc2a0cd5fd324eb3c5ada272ec64a902/src/buildx/inputs.ts#L114

I think we should remove `provenance: true` and explain the default behavior.

Reply:
I was not aware that it was handled differently depending on whether it's a public or private repository. We should probably mention that... in these docs 😆

You mean we shouldn't use the `true` shorthand but use a `mode=max` str value?

Reply:
> I was not aware that it was handled differently depending on whether it's a public or private repository. We should probably mention that... in these docs 😆

Yes sorry I forgot to mention that :see_no_evil:

> You mean we shouldn't use the `true` shorthand but use a `mode=max` str value?

I mean not defining it at all in this example as it would override the default behavior (https://github.com/docker/build-push-action/blob/4a13e500e55cf31b7a5d59a38ab2040ab0f42f56/src/context.ts#L155) that is, I think, a nice to have for handling provenance.

Reply:
OK. I think in the case of Docker Scout and using provenance attestations to help trace the source of their images, some orgs will definitely want to override the default behavior and push mode=max provenance for images in private repos. I guess we can have both examples and a description about it.

Reply:
Yes explaining the default case for provenance sounds good